### PR TITLE
Implement EZP-26375: Add siteaccess collector to debug toolbar

### PR DESCRIPTION
**Pull request by wizhippo**
> [EZP-26375](https://jira.ez.no/browse/EZP-26375)

### CODE REVIEW

File: eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
```
Comment (andrerom) [inline eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php]: can be skipped, we intend to remove these from all files.

**Comment by wizhippo** [comment]
You would be right.  So this should log siteaccess from the request attribute?  Or do I have to dig deeper and keep a log of changes on events too such as scope changes?

I guess as the purpose is to show what siteaccess the current request is using that would be the way to go?

File: eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php (hunk 1 above)
Comment (wizhippo) [inline eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php]: removed

**Comment by andrerom** [comment]
@bdunogier whats your thought here?

**Comment by bdunogier** [comment]
I would actually expect one siteaccess to show. Given how this toolbar is used (e.g. web context), users would expect most of the time the actually matched siteaccess to be used.

I would expect that the siteaccess will have been matched when debug data is collected, but I haven't tried.

**Comment by wizhippo** [comment]
@andrerom I think it is good to go as is and is inline with what I assumed and what @bdunogier commented.

**Comment by bdunogier** [comment]
Just tried it, works well.

Did you try adding details about the matcher in the panel ?

**Comment by wizhippo** [comment]
Beyond the siteaccess and the matching type what details would you like to see?  The info would vary based on each matcher and we would have to add different data to the collector accordingly.  Looking at the current matchers there is not much additional info available.  For example map based matchers really only expose getMapKey.  Is this information along the lines of what you were thinking?

**Comment by bdunogier** [comment]
> Beyond the siteaccess and the matching type what details would you like to see? The info would vary based on each matcher and we would have to add different data to the collector accordingly. Looking at the current matchers there is not much additional info available. For example map based matchers really only expose getMapKey. Is this information along the lines of what you were thinking?

If there is nothing worth displaying in the matcher itself, let's skip it. I haven't studied in details what is made available there based on the different matching types.

**Comment by wizhippo** [comment]
Sounds good.  I'll leave it as is

**Comment by bdunogier** [comment]
Approved, I'll send it to QA.

If you have a few spare minutes, you may add a screenshot to the PR' description (but you don't have to) ;)

File: eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php (hunk 1 above)
Comment (bdunogier) [inline eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php]: Actually, @andrerom only meant this particular line (the `@version` one), not the whole block. Can you restore it please ?

**Comment by bdunogier** [comment]
(still sent to QA, despite the change request. Very minor one)

File: eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php (hunk 1 above)
Comment (wizhippo) [inline eZ/Bundle/EzPublishDebugBundle/Collector/SiteAccessCollector.php]: done in https://github.com/ezsystems/ezpublish-kernel/pull/1790/commits/2a24de3f00a0d953265789def08931b1b2f9240e

**Pull request merged**